source/solvers/navier_stokes_base.cc

```diff
@@ -567,82 +572,242 @@ NavierStokesBase<dim, VectorType, DofsType>::finish_time_step()
+  // u_{n+1} = u_n + sum(b_i * k_i)
+  auto &sum_bi_ki      = this->sdirk_vectors.sum_bi_ki;
+  auto &temp_sum_bi_ki = this->sdirk_vectors.temp_sum_bi_ki;
+
+  // At each time iteration, we update the value of present_solution
+  temp_sum_bi_ki         = sum_bi_ki;
+  local_evaluation_point = previous_solutions[0];
+  local_evaluation_point.add(time_step, temp_sum_bi_ki);
+  present_solution = local_evaluation_point;
+}
+
 // Do an iteration with the NavierStokes Solver
 // Handles the fact that we may or may not be at a first
 // iteration with the solver and sets the initial condition
 template <int dim, typename VectorType, typename DofsType>
 void
 NavierStokesBase<dim, VectorType, DofsType>::iterate()
 {
+  // The locally_relevant_dofs are used for the assembly,
+  // while the non-locally_relevant_dofs are used for the calculus on the
+  // vectors (addition, multiplication, etc.).
+  const auto         method = this->simulation_control->get_assembly_method();
+  const auto         time_step = this->simulation_control->get_time_step();
+  const unsigned int n_stages = SimulationControl::get_number_of_stages(method);
+
   auto &present_solution = this->present_solution;
 
-  if (simulation_parameters.multiphysics.fluid_dynamics)
-    {
-      // Solve and percolate the auxiliary physics that should be treated BEFORE
-      // the fluid dynamics
-      multiphysics->solve(false,
-                          simulation_parameters.simulation_control.method);
-      multiphysics->percolate_time_vectors(false);
-
-      if (simulation_parameters.non_linear_solver.at(PhysicsID::fluid_dynamics)
-              .verbosity != Parameters::Verbosity::quiet ||
-          simulation_parameters.linear_solver.at(PhysicsID::fluid_dynamics)
-              .verbosity != Parameters::Verbosity::quiet)
-        announce_string(this->pcout, "Fluid Dynamics");
-      PhysicsSolver<VectorType>::solve_non_linear_system(false);
-
-      // If the physics need to be solved after the physics, the matrix free
-      // solver requires to update the value here. This is due to the different
-      // type of vectors.
-      if (this->multiphysics->get_active_physics().size() > 1)
-        this->update_solutions_for_multiphysics();
-
-      // Solve and percolate the auxiliary physics that should be treated AFTER
-      // the fluid dynamics
-      multiphysics->solve(true,
-                          simulation_parameters.simulation_control.method);
-      // Dear future Bruno, percolating auxiliary physics before fluid dynamics
-      // is necessary because of the checkpointing mechanism. You spent an
-      // evening debugging this, trust me.
-      multiphysics->percolate_time_vectors(true);
-    }
-  else
+  // The following variables are used for the SDIRK method
+  // u_{n+1} = u_n + sum(b_i * k_i)
+  auto &temp_sum_bi_ki = this->sdirk_vectors.temp_sum_bi_ki;
+  // At each time iteration, we reset the value of sum_bi_ki
+  temp_sum_bi_ki = 0;
```

Comment:
I would do this only if this is an sdirk method. You don,t need these tables and operation if it's not sdirk :).